pkg/output/output.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file contains functions used to implement the '--output' command line option.
+
+package output
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/ghodss/yaml"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"gitlab.com/c0b/go-ordered-json"
+)
+
+func Print(resource interface{}) {
+	var b bytes.Buffer
+	switch reflect.TypeOf(resource).String() {
+	case "[]*v1.CloudRegion":
+		if cloudRegions, ok := resource.([]*cmv1.CloudRegion); ok {
+			cmv1.MarshalCloudRegionList(cloudRegions, &b)
+		}
+	case "*v1.Cluster":
+		if cluster, ok := resource.(*cmv1.Cluster); ok {
+			cmv1.MarshalCluster(cluster, &b)
+		}
+	case "[]*v1.Cluster":
+		if clusters, ok := resource.([]*cmv1.Cluster); ok {
+			cmv1.MarshalClusterList(clusters, &b)
+		}
+	case "[]*v1.IdentityProvider":
+		if idps, ok := resource.([]*cmv1.IdentityProvider); ok {
+			cmv1.MarshalIdentityProviderList(idps, &b)
+		}
+	case "[]*v1.Ingress":
+		if ingresses, ok := resource.([]*cmv1.Ingress); ok {
+			cmv1.MarshalIngressList(ingresses, &b)
+		}
+	case "[]*v1.MachinePool":
+		if machinePools, ok := resource.([]*cmv1.MachinePool); ok {
+			cmv1.MarshalMachinePoolList(machinePools, &b)
+		}
+	case "[]*v1.MachineType":
+		if machineTypes, ok := resource.([]*cmv1.MachineType); ok {
+			cmv1.MarshalMachineTypeList(machineTypes, &b)
+		}
+	case "[]*v1.Version":
+		if versions, ok := resource.([]*cmv1.Version); ok {
+			cmv1.MarshalVersionList(versions, &b)
+		}
+	}
+	str, err := parseResource(b)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Seems odd to print the err if we can't parse the json/yaml. The command would exit `0` in this case which would be miss leading? Maybe we just want to exit `1` after this?

Reply:
Fixed. It doesn't feel right to call `os.Exit` outside of the command itself, so I decided to just return the error and check for it in every call to `output.Print`.